File: backend/app/core/phone.py
import re
from typing import Optional

def normalize_phone_number(phone: Optional[str]) -> Optional[str]:
    """
    Normalize phone numbers by:
    - Trimming whitespace
    - Removing internal spaces, hyphens, and parentheses
    - Preserving a leading '+' if present
    """
    if phone is None:
        return None
    trimmed = phone.strip()
    if not trimmed:
        return trimmed
    has_plus = trimmed.startswith("+")
    # Remove all non-digits
    cleaned = re.sub(r"\D", "", trimmed)
    if has_plus:
        return "+" + cleaned
    return cleaned

File: frontend/src/lib/phone.ts
/**
 * Normalizes phone numbers by:
 * - Trimming whitespace
 * - Removing internal spaces, hyphens, and parentheses
 * - Preserving a leading + if present
 */
export function normalizePhoneNumber(phone: string | null | undefined): string {
  if (!phone) return ''
  const trimmed = phone.trim()
  if (!trimmed) return ''
  const hasPlus = trimmed.startsWith('+')
  const cleaned = trimmed.replace(/[^0-9]/g, '')
  return hasPlus ? '+' + cleaned : cleaned
}
